ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetDataCommand.cpp

```diff
@@ -4,6 +4,373 @@
+	} //isAssembly
+	
+	
+	/*!
+	 Get the constituent parts of a specified element, e.g. the segments in a beam/column
+	 @param element The source element
+	 @param memo The source element memo
+	 @return An array of element parts (empty if the element is not made of parts - the original element is never included)
+	 */
+	auto getElementParts(const API_Element& element, const API_ElementMemo& memo) {
+		switch (element.header.type.typeID) {
+			case API_BeamID:
+				return getSegments(memo.beamSegments, [](const API_BeamSegmentType& segment, API_Element& element){ element.beamSegment = segment; });
+			case API_ColumnID:
+				return getSegments(memo.columnSegments, [](const API_ColumnSegmentType& segment, API_Element& element){ element.columnSegment = segment; });
+			default:
+				break;
+		}
+		return ElementArray{};
+	} //getElementParts
+	
+	
+	/*!
+	 Get the structure type for a specified element, e.g. how materials are arranged in the element body
+	 @param element The target element
+	 @return The element structural type
+	 */
+	auto getStructureType(const API_Element& element) {
+		switch (element.header.type.typeID) {
+			case API_BeamSegmentID:
+				return element.beamSegment.assemblySegmentData.modelElemStructureType;
+			case API_ColumnSegmentID:
+				return element.columnSegment.assemblySegmentData.modelElemStructureType;
+			case API_RoofID:
+				return element.roof.shellBase.modelElemStructureType;
+			case API_ShellID:
+				return element.shell.shellBase.modelElemStructureType;
+			case API_SlabID:
+				return element.slab.modelElemStructureType;
+			case API_WallID:
+				return element.wall.modelElemStructureType;
+			default:
+				return API_BasicStructure;
+		}
+	} //getStructureType
+	
+	
+	/*!
+	 Collect material quantities from the composite materials of a specified element
+	 @param element The target element to export the properties from
+	 @param elementQuantity Quantities extracted from the target element (out)
+	 @param extendedQuantity Optional extended quantities calculated for some element types
+	 @param quantityMask Mask to determine which quantities are required (minimise calculation time)
+	 @return An array of material quantities collected from the element composite structure
+	 */
+	auto measureQuantities(const API_Element& element, API_ElementQuantity& elementQuantity, API_Quantities& extendedQuantity,
+						   const API_QuantitiesMask& quantityMask) {
+		extendedQuantity.elements = &elementQuantity;
+		GS::Array<API_ElemPartQuantity> elementPartQuantities;
+		bool isPartQuantity = (element.header.type == API_MorphID);
+		extendedQuantity.elemPartQuantities = isPartQuantity ? &elementPartQuantities : nullptr;
+		API_QuantityPar quantityParameters{};
+		quantityParameters.minOpeningSize = Eps;
+		MaterialQuantArray result;
+		if (auto error = ACAPI_Element_GetQuantities(element.header.guid, &quantityParameters, &extendedQuantity, &quantityMask); error != NoError)
```

Comment:
We have a fun CI error related to using `init if statements`...
Something about C++17, lets have a chat, see if its as simple as adding the suggested compiler flag.

https://app.circleci.com/pipelines/github/specklesystems/speckle-sharp/16007/workflows/d16c1f09-8f11-4e56-a9fd-63feecca93a5/jobs/148736